concord-server/server/src/main/java/com/rabidgremlin/concord/functions/GetEligiblePhrasesForCompletionFunction.java

```diff
@@ -0,0 +1,97 @@
+package com.rabidgremlin.concord.functions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rabidgremlin.concord.api.Phrase;
+import com.rabidgremlin.concord.dao.model.GroupedPhraseVote;
+
+/**
+ * Function determines phrases which are eligible for completion. I.e. they have a label with vote count greater than
+ * the consensus AND a difference between the top two label vote counts less than consensus.
+ */
+public final class GetEligiblePhrasesForCompletionFunction
+{
+
+  private final List<GroupedPhraseVote> phraseVotes;
+
+  private final Map<String, String> phrasesVotedOnByResolver;
+
+  private final int consensusLevel;
+
+  private final Logger log = LoggerFactory.getLogger(GetEligiblePhrasesForCompletionFunction.class);
+
+  public GetEligiblePhrasesForCompletionFunction(List<GroupedPhraseVote> phraseVotes, Map<String, String> phrasesVotedOnByResolver, int consensusLevel)
+  {
+    this.phraseVotes = phraseVotes;
+    this.phrasesVotedOnByResolver = phrasesVotedOnByResolver;
+    this.consensusLevel = consensusLevel;
+  }
+
+  public List<Phrase> execute()
```

Comment:
Uuge method. Could do with breaking it up with a few private methods.

Reply:
i didnt write this, tempted to rewrite

Reply:
Your changes have made it too large to be readable though (~+20 lines), defs worth refactoring.